samples/hello-oboe/src/main/cpp/HelloOboeEngine.cpp

```diff
@@ -132,30 +132,39 @@ void HelloOboeEngine::restart() {
 
 oboe::Result HelloOboeEngine::start() {
     std::lock_guard<std::mutex> lock(mLock);
-    mIsLatencyDetectionSupported = false;
-    auto result = openPlaybackStream();
-    if (result == oboe::Result::OK){
-        mAudioSource =  std::make_shared<SoundGenerator>(mStream->getSampleRate(),
-                mStream->getChannelCount());
-        mLatencyCallback->setSource(std::dynamic_pointer_cast<IRenderableAudio>(mAudioSource));
-
-        LOGD("Stream opened: AudioAPI = %d, channelCount = %d, deviceID = %d",
-                 mStream->getAudioApi(),
-                 mStream->getChannelCount(),
-                 mStream->getDeviceId());
-
-        result = mStream->start();
-        if (result != oboe::Result::OK) {
-            LOGE("Error starting playback stream. Error: %s", oboe::convertToText(result));
-            mStream->close();
-            mStream.reset();
+    oboe::Result result = oboe::Result::OK;
+    // It is possible for a stream's device to become disconnected between the Open and the Start.
+    // So if it fails to start, close the old stream and try again.
+    int tryCount = 0;
+    bool startFailed = false;
+    do {
+        mIsLatencyDetectionSupported = false;
+        result = openPlaybackStream();
+        if (result == oboe::Result::OK){
+            mAudioSource =  std::make_shared<SoundGenerator>(mStream->getSampleRate(),
+                    mStream->getChannelCount());
+            mLatencyCallback->setSource(std::dynamic_pointer_cast<IRenderableAudio>(mAudioSource));
+
+            LOGD("Stream opened: AudioAPI = %d, channelCount = %d, deviceID = %d",
+                     mStream->getAudioApi(),
+                     mStream->getChannelCount(),
+                     mStream->getDeviceId());
+
+            result = mStream->requestStart();
+            if (result != oboe::Result::OK) {
+                LOGE("Error starting playback stream. Error: %s", oboe::convertToText(result));
+                mStream->close();
+                mStream.reset();
+                startFailed = true;
+            } else {
+                mIsLatencyDetectionSupported = (mStream->getTimestamp((CLOCK_MONOTONIC)) !=
+                                                oboe::Result::ErrorUnimplemented);
+                startFailed = false;
+            }
         } else {
-            mIsLatencyDetectionSupported = (mStream->getTimestamp((CLOCK_MONOTONIC)) !=
-                                            oboe::Result::ErrorUnimplemented);
+            LOGE("Error creating playback stream. Error: %s", oboe::convertToText(result));
```

Comment:
Should we add a startFailed = true here?

Reply:
I will check to make sure a DISCONNECT cannot occur during open.

Reply:
If a DISCONNECT occurs during open then it was possible for the stream to open and then the start would probably fail. AAudio was fixed so that the open could fail due to a disconnect.
See [b/193071260](http://b/193071260) | P2 | AAudio stream should not open OK if disconnected while opening

 I will retry if either the open or the start fails.